lib/elixir/src/elixir_exp.erl

```diff
@@ -230,8 +230,16 @@ expand({fn, Meta, Pairs}, E) ->
 
 expand({'cond', Meta, [KV]}, E) ->
   assert_no_match_or_guard_scope(Meta, 'cond', E),
-  {EClauses, EC} = elixir_exp_clauses:'cond'(Meta, KV, E),
-  {{'cond', Meta, [EClauses]}, EC};
+  {EBody, EC} = elixir_exp_clauses:'cond'(Meta, KV, E),
+  [{do, EClauses}] = EBody,
+  case lists:last(EClauses) of
```

Comment:
What if we move this to a separate function? `assert_no_underscore_clause_in_cond(EBody)`?